hipcheck/src/cache.rs

```diff
@@ -288,3 +333,28 @@ impl HcCache {
 		println!("{}", Table::new(to_show));
 	}
 }
+// This ensures the current state of the cache gets written to file. It is
+// important that a "HcCache" instance is dropped before execution terminates
+// so that this function gets a chance to run
```

Comment:
Hm, Rust doesn't guarantee destructors, so if the program is ended early (for example, a `SIGINT`) this `Drop` impl won't be called. If you _really_ want to make sure it's updated you'll need a signal handler. I don't think that's blocking for this though; could be done later.

Reply:
LogRocket has a decent guide to signal handling in Rust: https://blog.logrocket.com/guide-signal-handling-rust/